packages/react/src/data-connect/useConnectMutation.ts

```diff
@@ -0,0 +1,26 @@
+import { useMutation, type UseMutationResult } from "@tanstack/react-query";
+import {
+  executeMutation,
+  type MutationRef,
+  type DataConnect,
+} from "firebase/data-connect";
+
+export function useConnectMutation<
+  Fn extends (...args: any[]) => MutationRef<any, any>,
+  Data = ReturnType<Fn> extends MutationRef<infer D, any> ? D : never,
+  Variables = Fn extends (
+    dc: DataConnect,
+    vars: infer V
```

Comment:
allowing for DataConnect class instance and or just variables where the ref function can infer the variables directly given that the refs accommodate for this. for example this is a snapshot from the generated sdk for movies connector

<img width="1015" alt="image" src="https://github.com/user-attachments/assets/cfa1b1fb-2c1e-4027-aeb8-c85f05499c7c">